e2e-tests/src/test/button_game/mod.rs

```diff
@@ -159,6 +160,12 @@ pub fn simple_dex(config: &Config) -> Result<()> {
 
     dex.swap(account_conn, token1, balance_after, token3, mega(90))?;
     assert_recv_id(&mut events, "Swapped");
+
+    // can't swap a pair not on the whitelist
+
+    dex.remove_swap_pair(authority_conn, token3.into(), token1.into())?;
+    assert_recv_id(&mut events, "SwapPairRemoved");
```

Comment:
Why had the test below been failing before you added these two lines? i.e. it [the test] asserts that there was no trade made and I'm assuming that it used to pass so why did it?

Reply:
cause before pair was not on a whitelist for swapping